File: src/models/entities/Entity.java
package models.entities;

import models.body.Body;

/**
 * Abstract class for all entities in the game.
 * Includes minimum stats set, a body and a name
 */
public abstract class Entity {
    protected String name;
    protected Attribute strength;
    protected Attribute dexterity;
    protected Attribute endurance;
    protected Attribute intelligence;
    protected Attribute will;
    protected Attribute charisma;
    protected Attribute libido;
    protected Attribute sensitivity;
    protected Attribute lust;
    protected Body body;

    public Entity(String name) {
        this.name = name;
        strength = new Attribute(0, 100);
        dexterity = new Attribute(0, 100);
        endurance = new Attribute(0, 100);
        intelligence = new Attribute(0, 100);
        will = new Attribute(0, 100);
        charisma = new Attribute(0, 100);
        libido = new Attribute(0, 100);
        sensitivity = new Attribute(0, 100);
        lust = new Attribute(0, 100);
        body = new Body();
    }

    public String getName() {
        return name;
    }

    public Attribute getStrength() {
        return strength;
    }

    public Attribute getDexterity() {
        return dexterity;
    }

    public Attribute getEndurance() {
        return endurance;
    }

    public Attribute getIntelligence() {
        return intelligence;
    }

    public Attribute getWill() {
        return will;
    }

    public Attribute getCharisma() {
        return charisma;
    }

    public Attribute getLibido() {
        return libido;
    }

    public Attribute getSensitivity() {
        return sensitivity;
    }

    public Attribute getLust() {
        return lust;
    }

    public Body getBody() {
        return body;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Entity)) return false;

        Entity entity = (Entity) o;

        return name.equals(entity.name);

    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }
}

File: Assets/Scripts/entities/Entity.cs
using body;
using items;

namespace entities {

	/**
	 * Abstract class for all entities in the game.
	 * Includes minimum stats set, a body and a name
	 */
	public abstract class Entity {
	    protected string Name;
	    protected Attribute Strength;
	    protected Attribute Dexterity;
	    protected Attribute Endurance;
	    protected Attribute Intelligence;
	    protected Attribute Will;
	    protected Attribute Charisma;
	    protected Attribute Libido;
	    protected Attribute Sensitivity;
	    protected Attribute Lust;
	    protected Body Body;
        protected Backpack Backpack;

        public string name {
            get { return this.Name; }
        }
        public Attribute strength {
            get { return Strength; }
        }
        public Attribute dexterity {
            get { return Dexterity; }
        }
        public Attribute endurance {
            get { return Endurance; }
        }
        public Attribute intelligence {
            get { return Intelligence; }
        }
        public Attribute will {
            get { return Will; }
        }
        public Attribute charisma {
            get { return Charisma; }
        }
        public Attribute libido {
            get { return Libido; }
        }
        public Attribute sensitivity {
            get { return Sensitivity; }
        }
        public Attribute lust {
            get { return Lust; }
        }
        public Body body {
            get { return Body; }
        }
        public Backpack backpack {
            get { return Backpack; }
        }

        public Entity(string name, int backpackCapacity = 1) {
	        this.Name = name;
	        Strength = new Attribute(0, 100);
	        Dexterity = new Attribute(0, 100);
	        Endurance = new Attribute(0, 100);
	        Intelligence = new Attribute(0, 100);
	        Will = new Attribute(0, 100);
	        Charisma = new Attribute(0, 100);
	        Libido = new Attribute(0, 100);
	        Sensitivity = new Attribute(0, 100);
	        Lust = new Attribute(0, 100);
	        Body = new Body();
            this.Backpack = new Backpack(backpackCapacity);
	    }

	    public override bool Equals(System.Object obj) {
			if (this == obj) return true;
			if (!(obj is Entity)) return false;

	        Entity entity = (Entity) obj;

	        return Name.Equals(entity.Name);

	    }

	    public override int GetHashCode() {
	        return Name.GetHashCode();
	    }
	}
}
